### PR TITLE
fix(Popup|Tooltip): fix wrong positioning of pointing beak on content edges

### LINKED ISSUE #2365: Popup: pointing beak is broken on edges
## Bug Report

Sometimes pointing break is broken in `Popup`.

### Steps
1. Open https://codesandbox.io/s/fluent-ui-example-zo041

### Expected Result
Beak is positioned properly.

### Actual Result

image

### Version
0.44.0

### Testcase
https://codesandbox.io/s/fluent-ui-example-zo041

### PR DESCRIPTION
**Pull request by layershifter**
Fixes #2365.

This PR:
- fixes reported issues via `margin`s to have a gap between `pointer` and `content` edges
- adds Screener tests to catch potential regressions

### CODE REVIEW

File: packages/react/src/utils/positioner/Popper.tsx [hunk 1]
```diff
@@ -191,6 +192,7 @@ const Popper: React.FunctionComponent<PopperProps> = props => {
          * This modifier is necessary in order to render the pointer. Refs are resolved in effects, so it can't be
          * placed under computed modifiers. Deep merge is not required as this modifier has only these properties.
          */
+        keepTogether: { enabled: true },
```
Comment (layershifter) [inline packages/react/src/utils/positioner/Popper.tsx]: 💣 Needs #2361.

File: packages/react/src/themes/teams/getPointerStyles.ts [hunk 2]
```diff
@@ -29,21 +33,29 @@ const getPointerStyles = (
...
     right: {
       left: `-${pointerOffset}`,
+      marginBottom: pointerGap,
+      marginTop: pointerGap,
       transform: isSvg ? `rotate(${rtl ? 180 : 0}deg)` : 'rotate(135deg)',
     },
     bottom: {
       top: `-${pointerOffset}`,
+      marginLeft: pointerGap,
+      marginRight: pointerGap,
```
Comment (layershifter) [inline packages/react/src/themes/teams/getPointerStyles.ts:52]: `pointerGap` as variable name don't look intuitive, but I don't have better ideas

File: docs/src/examples/components/Popup/Types/PopupExamplePointerMargin.shorthand.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    />
+    <Popup
+      align="bottom"
+      content="Sample popup"
+      open
+      pointing
+      position="before"
+      trigger={<Box styles={{ border: '2px solid green' }}>X</Box>}
+    />
+  </Grid>
```
Comment (silviuaavram) [inline docs/src/examples/components/Popup/Types/PopupExamplePointerMargin.shorthand.tsx:102]: no need for `position="after"` examples?

**Review by silviuaavram — APPROVED** [review]
Just had one question

File: docs/src/examples/components/Popup/Types/PopupExamplePointerMargin.shorthand.tsx (hunk 3 above)
Comment (layershifter) [reply docs/src/examples/components/Popup/Types/PopupExamplePointerMargin.shorthand.tsx:102]: Reasonable, added 👍

**Pull request merged**